workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.ts

```diff
@@ -92,7 +97,62 @@ export class DynamicAuthorizationService {
   public async isAuthorizedOnSubject(
     isAuthorizedOnSubjectRequest: IsAuthorizedOnSubjectRequest
   ): Promise<void> {
-    throw new Error('Not implemented');
+    const validatedRequest = IsAuthorizedOnSubjectRequestParser.parse(isAuthorizedOnSubjectRequest);
+    const { authenticatedUser, dynamicOperation } = validatedRequest;
+    const metadata: Metadata = {
+      actor: authenticatedUser,
+      action: this.isAuthorizedOnSubject.name,
+      source: {
+        serviceName: DynamicAuthorizationService.name
+      },
+      requestBody: validatedRequest
+    };
+    try {
+      const { roles, id } = authenticatedUser;
+      const { subject, action } = dynamicOperation;
+      const { subjectId, subjectType } = subject;
+      //Create group identities
+      const identities: Identity[] = roles.map((groupId) => {
+        return {
+          identityType: 'GROUP',
+          identityId: groupId
+        };
+      });
+      //Add user identity
+      identities.push({
+        identityType: 'USER',
+        identityId: id
+      });
+      const identityPermissionsPromise = this._getAllIdentityPermissionsBySubject({
+        subjectId,
+        subjectType,
+        action,
+        identities
+      });
+      const identityPermissionsPromises = [identityPermissionsPromise];
+      if (subjectId !== this._wildcardSubjectId) {
+        const wildcardIdentityPermissionsPromise = this._getAllIdentityPermissionsBySubject({
+          subjectId: this._wildcardSubjectId,
+          subjectType,
+          action,
+          identities
+        });
+        identityPermissionsPromises.push(wildcardIdentityPermissionsPromise);
+      }
+      const [identityPermissions, wildcardIdentityPermissions] = await Promise.all(
+        identityPermissionsPromises
+      );
+      await this._authorizationPlugin.isAuthorizedOnDynamicOperations(
+        [...identityPermissions, ...(wildcardIdentityPermissions ?? [])],
+        [dynamicOperation]
+      );
+      metadata.statusCode = 200;
```

Comment:
HTTP Status codes should be defined at the controller level, not service level. Can we remove these?

Reply:
these are requirements of the audit service. We can remove the statusCode requirement and instead require a status of the operation. This would be less confusing